impl/src/main/java/io/opencensus/impl/internal/DisruptorEventQueue.java

```diff
@@ -143,6 +143,17 @@ public void enqueue(Entry entry) {
     }
   }
 
+  /**
+   * Shuts down the underlying disruptor.
+   *
+   * <p>Unfortunately there is no underlying public flush mechanism, without it there is a race
+   * condition in the ring buffer where it can hold events into the jvm shutdown.
+   */
+  @Override
+  public void shutdown() {
+    disruptor.shutdown();
```

Comment:
In theory, but this would require a check on every publish/enqueue call from what I can tell, unless we swap something out with a Noop impl. 

but this is a shutdown, its intended to be run during the jvm shutdown, so new spans should stop at some point naturally. thus this continuing to accept spans until there are no more might be considered a virtue. 

if the jvm never shuts down, there is clearly an issue elsewhere (the thing accepting requests that spawns spans is not itself shutdown).

Reply:
I meant that I don't think that this class should continue to enqueue Entries on the Disruptor after the Disruptor has been shut down.  That could cause the thread that recorded the instrumentation data to block forever.  I think it would be better to discard the data that cannot be processed, and log an error, to avoid interfering with the application.

I'm not sure how we could make this class continue to accept data after shutdown without a method supporting similar behavior on the Disruptor.

/cc @bogdandrutu

Reply:
I think that the number of entries is limited by the buffer size, but I want to avoid having the call to `enqueue` not return when the buffer is full.  I prefer your suggestion to swap the disruptor for a no-op version.  The no-op `enqueue` method could also log a warning the first time it is called.